server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportDesiredNodesActionsIT.java

```diff
@@ -383,7 +381,8 @@ public ClusterState execute(ClusterState currentState) throws Exception {
 
             @Override
             public void onFailure(Exception e) {
-
+                blockingClusterStateUpdateTaskExecuting.countDown();
```

Comment:
Recommend also bounding the wait time (`assertTrue(blockingClusterStateUpdateTaskExecuting.await(10, TimeUnit.SECONDS));` and similarly for `unblockClusterStateUpdateTask`).